packages/react-server/src/ReactFlightServer.js

```diff
@@ -3998,14 +4016,15 @@ export function stopFlowing(request: Request): void {
 
 // This is called to early terminate a request. It creates an error at all pending tasks.
 export function abort(request: Request, reason: mixed): void {
+  if (request.status === OPEN) {
+    request.status = ABORTING;
+  }
   try {
-    if (request.status === OPEN) {
-      request.status = ABORTING;
-    }
     const abortableTasks = request.abortableTasks;
     // We have tasks to abort. We'll emit one error row and then emit a reference
     // to that row from every row that's still remaining.
     if (abortableTasks.size > 0) {
+      request.status = ABORTING;
       request.pendingChunks++;
       const errorId = request.nextChunkId++;
       request.fatalError = errorId;
```

Comment:
Everything from here down is the same between abort/halt. So it seems like you could just have one big utility function that does the rest instead of splitting it into various smaller helpers that might be early abstraction.

Reply:
But really if we did encode Halt like we do Postpone as a separate row then they would be even more similar.